account-kit/infra/src/client/smartAccountClient.ts

```diff
@@ -174,21 +171,26 @@ export function createAlchemySmartAccountClient(
       ? alchemyUserOperationSimulator(config.transport)
       : undefined,
     signUserOperation: config.signUserOperation,
+    // @ts-expect-error
+    addBreadcrumb(crumb: string) {
+      const oldConfig = config.transport.config;
+      const dynamicFetchOptions = config.transport.dynamicFetchOptions;
+      const newTransport = alchemy({ ...oldConfig });
+      newTransport.updateHeaders(
+        headersUpdate(crumb)(
+          convertHeadersToObject(dynamicFetchOptions?.headers)
```

Comment:
now we update the headers to include any of the previously set headers (so during a given trace)